src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs

```diff
@@ -133,6 +218,37 @@ public static class ILoggerExtensions
             LogDependency(logger, dependencyType, dependencyData, isSuccessful, measurement.StartTime, measurement.Elapsed, context);
         }
 
+        /// <summary>
+        ///     Logs a dependency.
+        /// </summary>
+        /// <param name="logger">The logger to track the telemetry.</param>
+        /// <param name="dependencyType">Custom type of dependency</param>
+        /// <param name="dependencyData">Custom data of dependency</param>
+        /// <param name="isSuccessful">Indication whether or not the operation was successful</param>
+        /// <param name="dependencyName">The name of the dependency</param>
+        /// <param name="measurement">Measuring the latency to call the dependency</param>
+        /// <param name="context">Context that provides more insights on the dependency that was measured</param>
+        /// <exception cref="ArgumentNullException">
+        ///     Thrown when the <paramref name="logger"/>, <paramref name="dependencyData"/>, <paramref name="measurement"/> is <c>null</c>.
+        /// </exception>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="dependencyData"/> is blank.</exception>
+        public static void LogDependency(
+            this ILogger logger,            
+            string dependencyType,
+            object dependencyData,
+            bool isSuccessful,
+            string dependencyName,
+            DependencyMeasurement measurement,
```

Comment:
This feels more natural to me 
```suggestion
            string dependencyType,
            string dependencyName,
            object dependencyData,
            bool isSuccessful,
            DependencyMeasurement measurement,
```

Reply:
I had it like that initially but then I run into:
```The call is ambiguous between the following methods or properties: 'ILoggerExtensions.LogDependency(ILogger, string, string, object, bool, DateTimeOffset, TimeSpan, Dictionary<string, object>)' and 'ILoggerExtensions.LogDependency(ILogger, string, object, string, bool, DateTimeOffset, TimeSpan, Dictionary<string, object>)'```

Specifically these methods are ambiguous:
```
public static void LogDependency(
	this ILogger logger,            
	string dependencyType,
	string dependencyName,
	object dependencyData,
	bool isSuccessful,
	DateTimeOffset startTime,
	TimeSpan duration,
	Dictionary<string, object> context = null)
```
and 
```
public static void LogDependency(
	this ILogger logger,
	string dependencyType,
	object dependencyData,
	string targetName,
	bool isSuccessful,
	DateTimeOffset startTime,
	TimeSpan duration,
	Dictionary<string, object> context = null)
```

I figured to put the dependencyName as the last value before the DateTimeOffset/DependencyMeasurement as this will make the methods unique. Any other options to consider here?

Reply:
Dammit! No problem!